### PR TITLE
Only show (special) on profiles page when build is LVL3, F2P, F2P_LVL3

### PR DESCRIPTION
**Pull request by arvcork**
_Re: #1394_

**Changes:**

1. Introduce new method to check wherever you are in the application if the player has a special EHP rate. `hasSpecialEhp(player)`.
2. Used this to determine if the (Special) should show on the player details page.

**Evidence:**
Normal account (1 def pure):
Screenshot 2024-01-18 at 12 43 52

Special account (skiller):
Screenshot 2024-01-18 at 12 44 23

### CODE REVIEW

File: app/src/components/players/PlayerStatsTable.tsx [hunk 1]
```diff
@@ -214,7 +214,7 @@ function getSkillColumnDefinitions(player: Player, showVirtualLevels: boolean):
         return <TableSortButton column={column}>Skill</TableSortButton>;
       },
       cell: ({ row }) => {
-        const isSpecialEHP = player.type !== PlayerType.REGULAR || player.build !== PlayerBuild.MAIN;
+        const isSpecialEHP = hasSpecialEhp(player);
```
Comment (psikoi) [inline app/src/components/players/PlayerStatsTable.tsx:217]: It should be

`const isSpecialEHP = player.type !== PlayerType.REGULAR || hasSpecialEhp(player)`

File: app/src/utils/metrics.ts [hunk 2]
```diff
@@ -1,4 +1,8 @@
-import { COMBAT_SKILLS, MEMBER_SKILLS, Metric, PlayerBuild } from "@wise-old-man/utils";
+import { COMBAT_SKILLS, MEMBER_SKILLS, Metric, Player, PlayerBuild } from "@wise-old-man/utils";
+
+export function hasSpecialEhp(player: Player) {
+  return ['f2p', 'f2p_lvl3', 'lvl3'].includes(player.build);
```
Comment (psikoi) [inline app/src/utils/metrics.ts:4]: Could this just be inlined in `PlayerStatsTable`? It isn't used anywhere else

**Comment by arvcork** [comment]
@psikoi updated based on your feedback.

File: app/src/components/players/PlayerStatsTable.tsx [hunk 3]
```diff
@@ -207,20 +207,23 @@ function PlayerSkillsTable(
 }
 
 function getSkillColumnDefinitions(player: Player, showVirtualLevels: boolean): ColumnDef<SkillValue>[] {
+  const hasSpecialEhp = (player: Player) => {
```
Comment (psikoi) [inline app/src/components/players/PlayerStatsTable.tsx:210]: this doesn't need to be a function, I think just

`const hasSpecialEHP = player.type !== PlayerType || ['f2p', 'f2p_lvl3', 'lvl3'].includes(player.build);`

would suffice

Comment (arvcork) [reply app/src/components/players/PlayerStatsTable.tsx:210]: Hey @psikoi . Of course, I will do these changes tomorrow. Apologies for the delay!

**Comment by arvcork** [comment]
@psikoi Updated to reflect the changes you have mentioned +1

**Pull request merged**